lib/base-text-component.js

```diff
@@ -7,10 +7,11 @@ export default class BaseTextComponent extends Component {
         this.state = {
             type: props.type,
             value: '',
-            options: null
+            options: props.options
         };
 
         this._resolveMaskHandler();
+        this.state.value = this._getDefaultMaskedValue(props.value);
```

Comment:
Hi!
The state value is already on line `9`.
Could we move this part to there?

Reply:
The _getDefaultMaskedValue requires the _resolveMaskHandler method to be invoked first and the _resolveMaskedHandler requires the state type and options. I left the previous state.value assignment of '' because I think in theory it's possible that a mask requires an 'oldvalue' like money mask but that appears to be the only one and null checks anyway - so I could perhaps just delete line 9.